### PR TITLE
docs: add `border-radius` to `hX:target` selector styles

### PR DESCRIPTION
**Pull request by lumirlumir**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[X] Other, please explain:

#### What changes did you make? (Give an overview)

Hello,

I've resolved the issue mentioned in https://github.com/eslint/eslint.org/issues/662

> ### What problem do you want to solve?
> Hello, I noticed that when clicking the link button next to headings, the `background-color` changes, as shown in the screenshot below.
> 
> Image
> 
> However, I found that the `border-radius` for the `hX:target` selectors are not set to `var(--border-radius);`. This causes inconsistency in the visualization, as seen in the following screenshot.
> 
> Image
> 
> ### What do you think is the correct solution?
> I suggest adding a `border-radius: var(--border-radius);` property to the `hX:target` selectors to enhance visual consistency.
> 
> Here’s a preview of the changes:
> 
> * Before
>   Image
> * After
>   Image
> 
> ### Participation
> * [x]  I am willing to submit a pull request for this change.
> 
> ### Additional comments
> I think adding a bit more top and bottom padding or margin could improve the UI. Just my opinion—feel free to take it into consideration!

### CODE REVIEW

File: docs/src/assets/scss/foundations.scss [hunk 1]
```diff
@@ -10,6 +10,7 @@ h4:target,
 h5:target,
 h6:target {
     background-color: var(--lighter-background-color);
+    border-radius: var(--border-radius);
```
Comment (amareshsm) [inline docs/src/assets/scss/foundations.scss:13]: Could we also add some padding, like `padding: 3px 6px`?

Comment (lumirlumir) [reply docs/src/assets/scss/foundations.scss:13]: Sure, I'll check out various styles and leave comments!

**Comment by lumirlumir** [comment]
We can add `padding` to the `h1, h2, h3, h4, h5, h6` selectors, such as `padding: 0.25rem 0`, as shown in the screenshot below:

image

However, I don't think adding left and right padding is a good idea, as it can make the headings appear indented, as demonstrated in the screenshot below (using `padding: 0.25rem`):

image

Additionally, padding values cannot be applied to `hX:target` selectors without causing a layout shift when users click on the heading.

For these reasons, I believe using `padding: 0.25rem 0` or `padding: 4px 0` (or similar values) to the `h1, h2, h3, h4, h5, h6` selectors would be a better approach.

File: docs/src/assets/scss/foundations.scss (hunk 1 above)
Comment (lumirlumir) [reply docs/src/assets/scss/foundations.scss:13]: @amareshsm I've updated it.

**Comment by lumirlumir** [comment]
I've updated it.

**Review by Tanujkanti4441 — APPROVED** [review]
LGTM! thanks.

**Review by amareshsm — APPROVED** [review]
LGTM Thanks.

**Pull request merged**